WordPress/WordPressTest/ReaderSiteServiceTests.swift

```diff
@@ -0,0 +1,92 @@
+import XCTest
+import Nimble
+import OHHTTPStubs
+
+class ReaderSiteServiceTests: CoreDataTestCase {
+
+    private var service: ReaderSiteService!
+
+    override func setUp() {
+        let accountService = AccountService(coreDataStack: contextManager)
+        accountService.createOrUpdateAccount(withUsername: "testuser", authToken: "authtoken")
+        self.service = ReaderSiteService(coreDataStack: contextManager)
+
+        stub(condition: isHost("public-api.wordpress.com")) { request in
+            NSLog("[Warning] Received an unexpected request sent to \(String(describing: request.url))")
+            return HTTPStubsResponse(error: URLError(.notConnectedToInternet))
+        }
+        HTTPStubs.removeAllStubs()
+    }
+
+    func testFollowSiteByURL() {
+        stub(condition: isHost("test.blog")) { _ in
+            HTTPStubsResponse(data: Data(), statusCode: 200, headers: nil)
+        }
+        stub(condition: isPath("/rest/v1.1/sites/test.blog")) { _ in
+            HTTPStubsResponse(jsonObject: ["ID": 42], statusCode: 200, headers: nil)
+        }
+        stub(condition: isPath("/rest/v1.1/sites/42/follows/mine")) { _ in
+            HTTPStubsResponse(jsonObject: ["is_following": false], statusCode: 200, headers: nil)
+        }
+        stub(condition: isPath("/rest/v1.1/sites/42/follows/new")) { _ in
+            HTTPStubsResponse(jsonObject: [:], statusCode: 200, headers: nil)
+        }
+        stub(condition: isPath("/rest/v1.2/read/sites/42")) { _ in
+            HTTPStubsResponse(jsonObject: [
+                "feed_ID": 100,
+                "feed_URL": "https://test.blog/feed",
+                "post_count": 0,
+            ], statusCode: 200, headers: nil)
+        }
+        waitUntil { done in
+            self.service.followSite(by: URL(string: "https://test.blog")!) {
+                XCTAssertTrue(true)
+                done()
+            } failure: { _ in
+                XCTFail("")
+                done()
+            }
+        }
```

Comment:
Suggestions:

1. An empty line to separate the test setup (arrange) from the execution (act, assert)
2. I think we can get away with the `XCTAssertTrue(true)` and relay on the `done()` being called expectation in `waitUntil` for our assertion

```suggestion

        waitUntil { done in
            self.service.followSite(by: URL(string: "https://test.blog")!) {
                done()
            } failure: { _ in
                XCTFail()
            }
        }
```

An even more succinct version could be without the `XCTFail()` as well. I would find this option distracting though, because empty closure usually make me wonder "why is it even here in the first place?"

```suggestion

        waitUntil { done in
            self.service.followSite(by: URL(string: "https://test.blog")!) {
                done()
            } failure: { _ in }
        }
```

Reply:
Here's an example to show the test written in that way still does it's job

Option 1, with the explicit failure. This results in two failure reported by Xcode, which some might find redundant:

<img width="1393" alt="image" src="https://user-images.githubusercontent.com/1218433/221460644-fa27fec6-ac0c-4226-b1f2-533cf61bdd90.png">

Option 2, with the failure implicit in `done` not being called. This results in only one failure reported by Xcode, which is neat. However, I find the test code less clear:

<img width="1380" alt="image" src="https://user-images.githubusercontent.com/1218433/221462025-528f5c6d-1d30-4ace-8787-83df16bcd903.png">

Reply:
Thanks for pointing this out. I was too quick to make this PR. They weren't the initial draft changes, and weren't supposed to be checked in 🥲 . I've simplified (hopefully) the assertions in https://github.com/wordpress-mobile/WordPress-iOS/pull/20216/commits/a0f91e5a8b29895cf4f36facdbb02d0bceb83cda

Reply:
Again, I was too quick to push up changes, and [should've waited](https://github.com/wordpress-mobile/WordPress-iOS/pull/20216/commits/c8450e88aa18fb45f81d5ff48083653511c4c0d7).